src/components/AgreementFilters/AgreementFilters.js

```diff
@@ -14,45 +20,33 @@ const FILTERS = [
   'tags'
 ];
 
-export default class AgreementFilters extends React.Component {
-  static propTypes = {
-    activeFilters: PropTypes.object,
-    data: PropTypes.object.isRequired,
-    filterHandlers: PropTypes.object,
-  };
-
-  static defaultProps = {
-    activeFilters: {}
-  };
-
-  state = {
+export default function AgreementFilters({ activeFilters, data, filterHandlers }) {
+  const intl = useIntl();
+  const [filterState, setFilterState] = useState({
     agreementStatus: [],
     renewalPriority: [],
     isPerpetual: [],
-    tags: [],
-  }
+    tags: []
+  });
 
-  static getDerivedStateFromProps(props, state) {
+  useEffect(() => {
     const newState = {};
-
     FILTERS.forEach(filter => {
-      const values = props.data[`${filter}Values`];
-      if (values.length !== state[filter].length) {
+      const values = data[`${filter}Values`];
+      if (values.length !== filterState[filter]?.length) {
         newState[filter] = values;
       }
     });
 
-    if ((props.data?.tagsValues?.length ?? 0) !== state.tags.length) {
-      newState.tags = props.data.tagsValues.map(({ label }) => ({ value: label, label }));
+    if ((data?.tagsValues?.length ?? 0) !== filterState.tags?.length) {
+      newState.tags = data.tagsValues.map(({ label }) => ({ value: label, label }));
     }
+    if (Object.keys(newState).length) {
+      setFilterState(newState);
```

Comment:
This is causing an infinite loop right now, as the useEffect hook depends on filterState which is getting updated everytime you called `setFilterState` here